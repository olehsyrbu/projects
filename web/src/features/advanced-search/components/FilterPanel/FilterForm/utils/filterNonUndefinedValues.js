export function filterNonUndefinedValues(data) {
  return Object.keys(data).reduce((memo, key) => {
    if (data[key] !== undefined) {
      memo[key] = data[key];
    }
    return memo;
  }, {});
}
