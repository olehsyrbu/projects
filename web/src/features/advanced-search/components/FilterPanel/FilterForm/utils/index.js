export * from './filterNonUndefinedValues';
export * from './getFiltersCount';
export * from './toItems';
export * from './filterPanelPropTypes';
