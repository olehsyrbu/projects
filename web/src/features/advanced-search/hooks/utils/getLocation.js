import { isEmpty } from 'lodash-es';

export function getLocation({ locationCoordinates, bounds, region, needToAddBoundsInLocation }) {
  let location = {};
  const hasBounds =
    bounds &&
    bounds.northeast?.lat &&
    bounds.northeast?.lng &&
    bounds.southwest &&
    bounds.southwest?.lat &&
    bounds.southwest?.lng;

  if (hasBounds && needToAddBoundsInLocation) location.bounds = bounds;
  if (region) location.states = [region];
  if (!isEmpty(locationCoordinates)) location.coordinates = locationCoordinates;

  return location;
}
