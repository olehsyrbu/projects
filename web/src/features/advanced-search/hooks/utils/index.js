export * from './getLocation';
export * from './getSettings';
