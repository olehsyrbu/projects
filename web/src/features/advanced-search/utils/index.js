export * from './advancedSearchQueryProps';
export * from './hasLocationChangesWithinMap';
export * from './hasDifferentCoordinates';
export * from './cleanUpQuery';
export * from './prepForAnalytic';
export * from './hasNewLocation';
export * from './mapConstants';
export * from './getSort';
