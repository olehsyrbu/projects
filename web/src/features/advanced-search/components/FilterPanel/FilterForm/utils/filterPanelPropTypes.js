import { noop } from '@/core/utils';
import { SEARCH_CATEGORIES } from '@/modules/search/utils';
import PropType from 'prop-types';
import PropTypes from 'prop-types';

export function filterPanelPropTypes() {
  return {
    query: PropTypes.object,
    onApply: PropType.func,
    onClearAll: PropType.func,
    onClear: PropType.func,
    onDismiss: PropType.func,
    onChange: PropTypes.func,
    children: PropTypes.node,
  };
}

export function filterPanelDefaultProps() {
  return {
    query: {
      category: SEARCH_CATEGORIES.PROVIDER,
      modes: [],
    },
    onApply: noop,
    onClearAll: noop,
    onDismiss: noop,
    onClear: noop,
    onChange: noop,
  };
}
