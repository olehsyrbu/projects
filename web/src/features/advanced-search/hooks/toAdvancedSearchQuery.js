import { createSearchQueryConvertor } from './convertor';
import { AVAILABILITY_TYPES } from '@/core/definitions';
import { getLocation, getSettings } from './utils';

export const toAdvancedSearchQuery = createSearchQueryConvertor({
  modes: {
    type: Array,
    toQuery: (val) => ({
      modes: [...val],
    }),
  },
  page: {
    type: Number,
    default: 1,
  },
  limit: {
    type: Number,
    default: 5,
  },
  sort: {
    type: Object,
    toQuery: (val) => ({
      sort: val,
    }),
  },
  ageGroups: {
    type: Array,
    toQuery: (val) => ({
      age_groups: [...val],
    }),
  },
  accommodation: {
    type: Array,
    toQuery: (val) => ({
      accommodations: [...val],
    }),
  },
  amenities: {
    type: Array,
    toQuery: (val) => ({
      amenities: [...val],
    }),
  },
  dietaryAccommodations: {
    type: Array,
    toQuery: (val) => ({
      dietary_accommodation: [...val],
    }),
  },
  query: {
    type: String,
    toQuery: (val) => ({
      fuzzy_search: val,
    }),
  },
  specialties: {
    type: Array,
    toQuery: (val) => ({
      therapeutic_areas: [...val],
    }),
  },
  networkInsurance: {
    type: Array,
    toQuery: (val) => ({
      insurance_types: [...val],
    }),
  },
  paymentMethods: {
    type: Array,
    toQuery: (val) => ({
      payment_options: [...val],
    }),
  },
  programTypes: {
    type: Array,
    toQuery: (val) => ({
      program_types: [...val],
    }),
  },
  canAssistWithDailyLiving: {
    type: Boolean,
    toQuery: (val) => {
      if (!val) return;
      return {
        can_assist_with_daily_living: val,
      };
    },
  },
  treatsMedicallyUnstable: {
    type: Boolean,
    toQuery: (val) => {
      if (!val) return;
      return {
        treats_medically_unstable: val,
      };
    },
  },
  treatsSuicidalIdeation: {
    type: Boolean,
    toQuery: (val) => {
      if (!val) return;
      return {
        treats_suicidal_ideation: val,
      };
    },
  },
  rainbowMember: {
    type: Boolean,
    toQuery: (val) => {
      if (!val) {
        return;
      }
      return {
        rainbow_member: val,
      };
    },
  },
  sexualIdentities: {
    type: Array,
    toQuery: (val) => ({
      sexual_identities: [...val],
    }),
  },
  genders: {
    type: Array,
    toQuery: (val) => ({
      genders: [...val],
    }),
  },
  religions: {
    type: Array,
    toQuery: (val) => ({
      religions: [...val],
    }),
  },
  languages: {
    type: Array,
    toQuery: (val) => ({
      language_services: [...val],
    }),
  },
  ethnicities: {
    type: Array,
    toQuery: (val) => ({
      ethnicities: [...val],
    }),
  },
  modalities: {
    type: Array,
    toQuery: (val) => ({
      therapies: [...val],
    }),
  },
  setting: {
    type: String,
    toQuery: (setting, key, rest) => {
      return {
        OR: getSettings(setting),
        location: getLocation(rest),
      };
    },
  },
  location: {
    type: Object,
    toQuery: (_, key, rest) => {
      return {
        location: getLocation(rest),
      };
    },
  },
  treatments: {
    type: Array,
    toQuery: (val) => ({
      treatments: [...val],
    }),
  },
  organizationSubdomain: {
    type: String,
    toQuery: (val) => ({
      organization_subdomains: [val],
    }),
  },
  providerTypes: {
    type: Array,
    toQuery: (val) => ({
      provider_types: [...val],
    }),
  },
  canPrescribeMedication: {
    type: Boolean,
    toQuery: (val) => {
      if (!val) {
        return;
      }
      return {
        can_prescribe_medication: val,
      };
    },
  },
  isAcceptingNewClients: {
    type: Boolean,
    toQuery: (val) => {
      if (!val) {
        return;
      }
      return {
        availability: {
          status: {
            in: [AVAILABILITY_TYPES.ACCEPTING_NEW_CLIENTS, AVAILABILITY_TYPES.PLEASE_INQUIRE],
          },
        },
      };
    },
  },
  timeZone: {
    type: String,
    toQuery: (val) => ({
      availability: {
        timezone: val,
      },
    }),
  },
  startTime: {
    type: String,
    toQuery: (val) => ({
      availability: {
        start: val,
      },
    }),
  },
  endTime: {
    type: String,
    toQuery: (val) => ({
      availability: {
        end: val,
      },
    }),
  },
  day: {
    type: Array,
    toQuery: (val) => {
      return {
        availability: {
          day: val,
        },
      };
    },
  },
  active: {
    type: Boolean,
    default: true,
  },
  organizationTypes: {
    type: Object,
    toQuery: (val) => ({ organization_types: val }),
  },
  specialGroups: {
    type: Array,
    toQuery: (val) => ({
      special_groups: [...val],
    }),
  },
  hasPsypact: {
    type: Boolean,
    toQuery: (val) => {
      if (!val) {
        return;
      }
      return {
        has_psypact: val,
      };
    },
  },
});
