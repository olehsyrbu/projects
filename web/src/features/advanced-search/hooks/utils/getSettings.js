import { SEARCH_WORK_SETTING } from '@/modules/search/utils';

export function getSettings(setting) {
  const available = { available: true };
  let settings = [{ in_person: available }, { remote: available }];

  switch (setting) {
    case SEARCH_WORK_SETTING.REMOTE:
      settings = [{ remote: available }];
      break;
    case SEARCH_WORK_SETTING.IN_PERSON:
      settings = [{ in_person: available }];
      break;
    default:
      return settings;
  }
  return settings;
}
