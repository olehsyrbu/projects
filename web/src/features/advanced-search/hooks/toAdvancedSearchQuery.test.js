import { toAdvancedSearchQuery } from './toAdvancedSearchQuery';
import { SEARCH_WORK_SETTING } from '@/modules/search/utils';

it('has query into filter', () => {
  let searchText = 'bbkbkkbkbkbkb';

  expect(
    toAdvancedSearchQuery({
      query: searchText,
    }),
  ).toMatchSnapshot();
});

it('has specialties into filter', () => {
  const specialties = ['000', '111', '222'];

  expect(
    toAdvancedSearchQuery({
      specialties,
    }),
  ).toMatchSnapshot();
});

it('has specialGroups into filter', () => {
  const specialGroups = ['000', '111', '222'];

  expect(
    toAdvancedSearchQuery({
      specialGroups,
    }),
  ).toMatchSnapshot();
});

it('has organizationSubdomain into filter', () => {
  const organizationSubdomain = 'bcbsks';

  expect(
    toAdvancedSearchQuery({
      organizationSubdomain,
    }),
  ).toMatchSnapshot();
});

it('has day into filter', () => {
  const day = ['SUNDAY', 'MONDAY'];

  expect(
    toAdvancedSearchQuery({
      day,
    }),
  ).toMatchSnapshot();
});

it('provides default pagination', () => {
  expect(toAdvancedSearchQuery()).toEqual({
    filter: {
      active: true,
    },
    page: 1,
    limit: 5,
  });
});

it('provides age group', () => {
  const ageGroups = ['blblblbl', 'ssss'];
  expect(
    toAdvancedSearchQuery({
      ageGroups,
    }),
  ).toMatchSnapshot();
});

it('provides startTime', () => {
  const startTime = '12:30';
  expect(
    toAdvancedSearchQuery({
      startTime,
    }),
  ).toMatchSnapshot();
});

it('provides endTime', () => {
  const endTime = '15:30';
  const params = toAdvancedSearchQuery({
    endTime,
  });

  expect(params).toMatchSnapshot();
});

it('provides timezone', () => {
  const timeZone = 'Pacific/Pago_Pago';
  expect(
    toAdvancedSearchQuery({
      timeZone,
    }),
  ).toMatchSnapshot();
});

it('provides network insurance', () => {
  expect(
    toAdvancedSearchQuery({
      networkInsurance: ['000', '111'],
    }),
  ).toMatchSnapshot();
});

it('provides accommodations insurance', () => {
  expect(
    toAdvancedSearchQuery({
      accommodation: ['000', '111'],
    }),
  ).toMatchSnapshot();
});

it('provides payment methods', () => {
  expect(
    toAdvancedSearchQuery({
      paymentMethods: ['000', '111'],
    }),
  ).toMatchSnapshot();
});

it('provides modalities', () => {
  expect(
    toAdvancedSearchQuery({
      modalities: ['000', '111'],
    }),
  ).toMatchSnapshot();
});

it('provides treatments', () => {
  expect(
    toAdvancedSearchQuery({
      treatments: ['000', '111'],
    }),
  ).toMatchSnapshot();
});

it('converts location into filter', () => {
  let location = {
    lat: 11.0001,
    lng: 20.2222,
    distance: {
      max: 10000,
    },
    region: 'CA',
  };

  expect(
    toAdvancedSearchQuery({
      setting: 'inPerson',
      location,
    }),
  ).toMatchSnapshot();
});

it('converts remote only', () => {
  expect(
    toAdvancedSearchQuery({
      setting: 'remote',
    }),
  ).toMatchSnapshot();
});

it('converts both only', () => {
  expect(
    toAdvancedSearchQuery({
      setting: 'both',
    }),
  ).toMatchSnapshot();
});

it('converts in-person only', () => {
  expect(
    toAdvancedSearchQuery({
      setting: 'inPerson',
    }),
  ).toMatchSnapshot();
});

it('has to gql variables location with bounds', () => {
  expect(
    toAdvancedSearchQuery({
      query: 'Test Query',
      setting: 'inPerson',
      location: {
        bounds: {
          northeast: {
            lat: 85.9439285948254,
            lng: -45.486148925235454,
          },
          southwest: {
            lat: -59.58664350008189,
            lng: -126.69708642523545,
          },
        },
      },
    }),
  ).toMatchSnapshot();
});

it('has to gql variables location with bounds both', () => {
  expect(
    toAdvancedSearchQuery({
      query: 'Test Query',
      setting: 'both',
      location: {
        bounds: {
          northeast: {
            lat: 85.9439285948254,
            lng: -45.486148925235454,
          },
          southwest: {
            lat: -59.58664350008189,
            lng: -126.69708642523545,
          },
        },
      },
    }),
  ).toMatchSnapshot();
});

it('has to gql variables location and inPerson', () => {
  expect(
    toAdvancedSearchQuery({
      query: 'Test Query',
      setting: 'inPerson',
      location: {
        lat: 11.0001,
        lng: 20.2222,
        distance: {
          max: 10000,
        },
        region: 'TX',
      },
      specialties: ['000', '111', '222'],
    }),
  ).toMatchSnapshot();
});

it('has "rainbowMember"', () => {
  expect(
    toAdvancedSearchQuery({
      rainbowMember: true,
    }),
  ).toMatchSnapshot();

  expect(
    toAdvancedSearchQuery({
      rainbowMember: false,
    }),
  ).toMatchSnapshot();
});

it('has "genders"', () => {
  expect(
    toAdvancedSearchQuery({
      genders: ['dfdf', '4444', '494949494'],
    }),
  ).toMatchSnapshot();
});

it('has "religions"', () => {
  expect(
    toAdvancedSearchQuery({
      religions: ['dfdf', '4444', '494949494'],
    }),
  ).toMatchSnapshot();
});

it('has "languages"', () => {
  expect(
    toAdvancedSearchQuery({
      languages: ['dfdf', '4444', '494949494'],
    }),
  ).toMatchSnapshot();
});

it('has "ethnicities"', () => {
  expect(
    toAdvancedSearchQuery({
      ethnicities: ['dfdf', '4444', '494949494'],
    }),
  ).toMatchSnapshot();
});

it('has to gql variables location and remote', () => {
  expect(
    toAdvancedSearchQuery({
      query: 'Test Query',
      setting: 'remote',
      location: {
        lat: 11.0001,
        lng: 20.2222,
        distance: {
          max: 10000,
        },
        region: 'TX',
      },
      specialties: ['000', '111', '222'],
      page: 2,
      limit: 10,
    }),
  ).toMatchSnapshot();
});

it('has to gql variables location and both', () => {
  expect(
    toAdvancedSearchQuery({
      query: 'Test Query',
      setting: 'both',
      location: {
        lat: 11.0001,
        lng: 20.2222,
        distance: {
          max: 10000,
        },
        region: 'TX',
      },
      specialties: ['000', '111', '222'],
    }),
  ).toMatchSnapshot();
});

it('has to gql variables no location, no settings', () => {
  expect(
    toAdvancedSearchQuery({
      query: 'Test Query',
      specialties: ['000', '111', '222'],
    }),
  ).toMatchSnapshot();
});

it('has provider types', () => {
  expect(
    toAdvancedSearchQuery({
      providerTypes: ['000', '111', '222'],
    }),
  ).toMatchSnapshot();
});

it('has modes', () => {
  expect(
    toAdvancedSearchQuery({
      modes: ['PERSON', 'PROGRAM'],
    }),
  ).toMatchSnapshot();
});

it('has sexualIdentities', () => {
  expect(
    toAdvancedSearchQuery({
      sexualIdentities: ['123'],
    }),
  ).toMatchSnapshot();
});

it('has "can prescribe medication"', () => {
  expect(
    toAdvancedSearchQuery({
      canPrescribeMedication: true,
    }),
  ).toMatchSnapshot();

  expect(
    toAdvancedSearchQuery({
      canPrescribeMedication: false,
    }),
  ).toMatchSnapshot();
});

it('has "current availability"', () => {
  expect(
    toAdvancedSearchQuery({
      isAcceptingNewClients: true,
    }),
  ).toMatchSnapshot();

  expect(
    toAdvancedSearchQuery({
      isAcceptingNewClients: false,
    }),
  ).toMatchSnapshot();
});

it('has "canAssistWithDailyLiving"', () => {
  expect(
    toAdvancedSearchQuery({
      canAssistWithDailyLiving: true,
    }),
  ).toMatchSnapshot();

  expect(
    toAdvancedSearchQuery({
      canAssistWithDailyLiving: false,
    }),
  ).toMatchSnapshot();
});

it('has "treatsMedicallyUnstable"', () => {
  expect(
    toAdvancedSearchQuery({
      treatsMedicallyUnstable: true,
    }),
  ).toMatchSnapshot();

  expect(
    toAdvancedSearchQuery({
      treatsMedicallyUnstable: false,
    }),
  ).toMatchSnapshot();
});

it('has "treatsSuicidalIdeation"', () => {
  expect(
    toAdvancedSearchQuery({
      treatsSuicidalIdeation: true,
    }),
  ).toMatchSnapshot();

  expect(
    toAdvancedSearchQuery({
      treatsSuicidalIdeation: false,
    }),
  ).toMatchSnapshot();
});

it('has "amenities"', () => {
  expect(
    toAdvancedSearchQuery({
      amenities: ['1'],
    }),
  ).toMatchSnapshot();

  expect(
    toAdvancedSearchQuery({
      amenities: [],
    }),
  ).toMatchSnapshot();
});

it('has "dietaryAccommodations"', () => {
  expect(
    toAdvancedSearchQuery({
      dietaryAccommodations: ['1'],
    }),
  ).toMatchSnapshot();

  expect(
    toAdvancedSearchQuery({
      dietaryAccommodations: [],
    }),
  ).toMatchSnapshot();
});

it('has all params', () => {
  expect(
    toAdvancedSearchQuery({
      canAssistWithDailyLiving: true,
      treatsMedicallyUnstable: true,
      treatsSuicidalIdeation: true,
      modes: ['PERSON', 'PROGRAM'],
      dietaryAccommodations: ['1', '2'],
      amenities: ['am1', 'am2'],
      specialties: ['sp1', 'sp2'],
      sexualIdentities: ['sp1', 'sp2'],
      modalities: ['md1', 'md2'],
      treatments: ['tr1', 'tr2'],
      services: ['ser1', 'ser2'],
      programs: ['pr1', 'pr2'],
      providerTypes: ['pt1', 'pt2'],
      canPrescribeMedication: true,
      isAcceptingNewClients: true,
      location: {
        lat: 11.0001,
        lng: 20.2222,
        distance: {
          max: 10000,
        },
        region: 'TX',
      },
      setting: SEARCH_WORK_SETTING.BOTH,
      networkInsurance: ['nI1', 'nI2'],
      paymentMethods: ['pm1', 'pm2'],
      query: 'some name',
      specialGroups: ['sg1', 'sg2'],
    }),
  ).toMatchSnapshot();
});

it('provides sorting', () => {
  const sort = { nearest: 'ASC' };
  expect(
    toAdvancedSearchQuery({
      sort,
    }),
  ).toMatchSnapshot();
});

it('has special groups', () => {
  expect(
    toAdvancedSearchQuery({
      specialGroups: ['000', '111', '222'],
    }),
  ).toMatchSnapshot();
});
