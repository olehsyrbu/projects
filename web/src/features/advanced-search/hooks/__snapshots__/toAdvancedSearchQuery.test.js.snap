// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`converts both only 1`] = `
{
  "filter": {
    "OR": [
      {
        "in_person": {
          "available": true,
        },
      },
      {
        "remote": {
          "available": true,
        },
      },
    ],
    "active": true,
    "location": {},
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`converts in-person only 1`] = `
{
  "filter": {
    "OR": [
      {
        "in_person": {
          "available": true,
        },
      },
    ],
    "active": true,
    "location": {},
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`converts location into filter 1`] = `
{
  "filter": {
    "OR": [
      {
        "in_person": {
          "available": true,
        },
      },
    ],
    "active": true,
    "location": {},
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`converts remote only 1`] = `
{
  "filter": {
    "OR": [
      {
        "remote": {
          "available": true,
        },
      },
    ],
    "active": true,
    "location": {},
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "amenities" 1`] = `
{
  "filter": {
    "active": true,
    "amenities": [
      "1",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "amenities" 2`] = `
{
  "filter": {
    "active": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "can prescribe medication" 1`] = `
{
  "filter": {
    "active": true,
    "can_prescribe_medication": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "can prescribe medication" 2`] = `
{
  "filter": {
    "active": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "canAssistWithDailyLiving" 1`] = `
{
  "filter": {
    "active": true,
    "can_assist_with_daily_living": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "canAssistWithDailyLiving" 2`] = `
{
  "filter": {
    "active": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "current availability" 1`] = `
{
  "filter": {
    "active": true,
    "availability": {
      "status": {
        "in": [
          "ACCEPTING_NEW_CLIENTS",
          "PLEASE_INQUIRE",
        ],
      },
    },
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "current availability" 2`] = `
{
  "filter": {
    "active": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "dietaryAccommodations" 1`] = `
{
  "filter": {
    "active": true,
    "dietary_accommodation": [
      "1",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "dietaryAccommodations" 2`] = `
{
  "filter": {
    "active": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "ethnicities" 1`] = `
{
  "filter": {
    "active": true,
    "ethnicities": [
      "dfdf",
      "4444",
      "494949494",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "genders" 1`] = `
{
  "filter": {
    "active": true,
    "genders": [
      "dfdf",
      "4444",
      "494949494",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "languages" 1`] = `
{
  "filter": {
    "active": true,
    "language_services": [
      "dfdf",
      "4444",
      "494949494",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "rainbowMember" 1`] = `
{
  "filter": {
    "active": true,
    "rainbow_member": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "rainbowMember" 2`] = `
{
  "filter": {
    "active": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "religions" 1`] = `
{
  "filter": {
    "active": true,
    "religions": [
      "dfdf",
      "4444",
      "494949494",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "treatsMedicallyUnstable" 1`] = `
{
  "filter": {
    "active": true,
    "treats_medically_unstable": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "treatsMedicallyUnstable" 2`] = `
{
  "filter": {
    "active": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "treatsSuicidalIdeation" 1`] = `
{
  "filter": {
    "active": true,
    "treats_suicidal_ideation": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has "treatsSuicidalIdeation" 2`] = `
{
  "filter": {
    "active": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has all params 1`] = `
{
  "filter": {
    "OR": [
      {
        "in_person": {
          "available": true,
        },
      },
      {
        "remote": {
          "available": true,
        },
      },
    ],
    "active": true,
    "amenities": [
      "am1",
      "am2",
    ],
    "availability": {
      "status": {
        "in": [
          "ACCEPTING_NEW_CLIENTS",
          "PLEASE_INQUIRE",
        ],
      },
    },
    "can_assist_with_daily_living": true,
    "can_prescribe_medication": true,
    "dietary_accommodation": [
      "1",
      "2",
    ],
    "fuzzy_search": "some name",
    "insurance_types": [
      "nI1",
      "nI2",
    ],
    "location": {},
    "modes": [
      "PERSON",
      "PROGRAM",
    ],
    "payment_options": [
      "pm1",
      "pm2",
    ],
    "provider_types": [
      "pt1",
      "pt2",
    ],
    "sexual_identities": [
      "sp1",
      "sp2",
    ],
    "special_groups": [
      "sg1",
      "sg2",
    ],
    "therapeutic_areas": [
      "sp1",
      "sp2",
    ],
    "therapies": [
      "md1",
      "md2",
    ],
    "treatments": [
      "tr1",
      "tr2",
    ],
    "treats_medically_unstable": true,
    "treats_suicidal_ideation": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has day into filter 1`] = `
{
  "filter": {
    "active": true,
    "availability": {
      "day": [
        "SUNDAY",
        "MONDAY",
      ],
    },
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has modes 1`] = `
{
  "filter": {
    "active": true,
    "modes": [
      "PERSON",
      "PROGRAM",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has organizationSubdomain into filter 1`] = `
{
  "filter": {
    "active": true,
    "organization_subdomains": [
      "bcbsks",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has provider types 1`] = `
{
  "filter": {
    "active": true,
    "provider_types": [
      "000",
      "111",
      "222",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has query into filter 1`] = `
{
  "filter": {
    "active": true,
    "fuzzy_search": "bbkbkkbkbkbkb",
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has sexualIdentities 1`] = `
{
  "filter": {
    "active": true,
    "sexual_identities": [
      "123",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has special groups 1`] = `
{
  "filter": {
    "active": true,
    "special_groups": [
      "000",
      "111",
      "222",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has specialGroups into filter 1`] = `
{
  "filter": {
    "active": true,
    "special_groups": [
      "000",
      "111",
      "222",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has specialties into filter 1`] = `
{
  "filter": {
    "active": true,
    "therapeutic_areas": [
      "000",
      "111",
      "222",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has to gql variables location and both 1`] = `
{
  "filter": {
    "OR": [
      {
        "in_person": {
          "available": true,
        },
      },
      {
        "remote": {
          "available": true,
        },
      },
    ],
    "active": true,
    "fuzzy_search": "Test Query",
    "location": {},
    "therapeutic_areas": [
      "000",
      "111",
      "222",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has to gql variables location and inPerson 1`] = `
{
  "filter": {
    "OR": [
      {
        "in_person": {
          "available": true,
        },
      },
    ],
    "active": true,
    "fuzzy_search": "Test Query",
    "location": {},
    "therapeutic_areas": [
      "000",
      "111",
      "222",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has to gql variables location and remote 1`] = `
{
  "filter": {
    "OR": [
      {
        "remote": {
          "available": true,
        },
      },
    ],
    "active": true,
    "fuzzy_search": "Test Query",
    "location": {},
    "therapeutic_areas": [
      "000",
      "111",
      "222",
    ],
  },
  "limit": 10,
  "page": 2,
}
`;

exports[`has to gql variables location with bounds 1`] = `
{
  "filter": {
    "OR": [
      {
        "in_person": {
          "available": true,
        },
      },
    ],
    "active": true,
    "fuzzy_search": "Test Query",
    "location": {},
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has to gql variables location with bounds both 1`] = `
{
  "filter": {
    "OR": [
      {
        "in_person": {
          "available": true,
        },
      },
      {
        "remote": {
          "available": true,
        },
      },
    ],
    "active": true,
    "fuzzy_search": "Test Query",
    "location": {},
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`has to gql variables no location, no settings 1`] = `
{
  "filter": {
    "active": true,
    "fuzzy_search": "Test Query",
    "therapeutic_areas": [
      "000",
      "111",
      "222",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`provides accommodations insurance 1`] = `
{
  "filter": {
    "accommodations": [
      "000",
      "111",
    ],
    "active": true,
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`provides age group 1`] = `
{
  "filter": {
    "active": true,
    "age_groups": [
      "blblblbl",
      "ssss",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`provides endTime 1`] = `
{
  "filter": {
    "active": true,
    "availability": {
      "end": "15:30",
    },
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`provides modalities 1`] = `
{
  "filter": {
    "active": true,
    "therapies": [
      "000",
      "111",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`provides network insurance 1`] = `
{
  "filter": {
    "active": true,
    "insurance_types": [
      "000",
      "111",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`provides payment methods 1`] = `
{
  "filter": {
    "active": true,
    "payment_options": [
      "000",
      "111",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`provides sorting 1`] = `
{
  "filter": {
    "active": true,
  },
  "limit": 5,
  "page": 1,
  "sort": {
    "nearest": "ASC",
  },
}
`;

exports[`provides startTime 1`] = `
{
  "filter": {
    "active": true,
    "availability": {
      "start": "12:30",
    },
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`provides timezone 1`] = `
{
  "filter": {
    "active": true,
    "availability": {
      "timezone": "Pacific/Pago_Pago",
    },
  },
  "limit": 5,
  "page": 1,
}
`;

exports[`provides treatments 1`] = `
{
  "filter": {
    "active": true,
    "treatments": [
      "000",
      "111",
    ],
  },
  "limit": 5,
  "page": 1,
}
`;
